Source/WebKit/UIProcess/RemoteLayerTree/mac/RemoteLayerTreeDrawingAreaProxyMac.mm

```diff
@@ -206,20 +208,38 @@ explicit RemoteLayerTreeDisplayLinkClient(WebPageProxyIdentifier pageID)
 
 void RemoteLayerTreeDrawingAreaProxyMac::commitTransientZoom(double scale, FloatPoint origin)
 {
-    LOG_WITH_STREAM(ViewGestures, stream << "RemoteLayerTreeDrawingAreaProxyMac::commitTransientZoom - scale " << scale << " origin " << origin);
-
+    auto rootNode = downcast<ScrollingTreeFrameScrollingNode>(m_webPageProxy.scrollingCoordinatorProxy()->rootNode());
+    auto scrollPosition = rootNode->currentScrollPosition();
+    auto visibleContentRect = rootNode->layoutViewport();
```

Comment:
These two things are not the same. The layout viewport is larger than the visible content rect when zoomed in, so I don't think this is the right rectangle. I would expect some visible UI issues with this code.

Reply:
Yeah I was a bit confused by this but when I tried out the TiledCoreAnimationDrawingArea code I saw that the `frameView.visibleContentRectIncludingScrollbars` was always returning the viewport size, irrespective of scale. This is because `FrameView::visibleContentScaleFactor()` always returns 1 in this case. It is possible that this is a bug, but in both cases (UI-side compositing off and on) I am seeing the snapping behavior being properly prevented. At the very least it is improperly named.